src/typing/typeload.ml

```diff
@@ -121,36 +121,72 @@ let find_type_in_current_module_context ctx pack name =
 		ImportHandling.mark_import_position ctx pi;
 		t
 
-let load_unqualified_type_def ctx mname tname p =
-	try
-		let rec loop l =
-			match l with
-			| [] ->
-				raise Exit
-			| (pack,ppack) :: l ->
-				begin try
-					let mt = load_type ctx (pack,mname) tname p in
-					ImportHandling.mark_import_position ctx ppack;
-					mt
-				with Not_found ->
-					loop l
-				end
-		in
-		(* Check wildcard packages by using their package *)
-		loop ctx.m.wildcard_packages
-	with Exit ->
-		let rec loop l =
-			match l with
-			| [] ->
-				load_type_raise ctx ([],mname) tname p
-			| _ :: sl as l ->
+let find_in_wildcard_imports ctx mname p f =
+	let rec loop l =
+		match l with
+		| [] ->
+			raise Not_found
+		| (pack,ppack) :: l ->
+			begin
+			try
+				let path = (pack,mname) in
+				let m =
+					try
+						ctx.g.do_load_module ctx path p
+					with Error (Module_not_found mpath,_) when mpath = path ->
+						raise Not_found
+				in
+				let r = f m ~resume:true in
```

Comment:
I'm a bit worried that this relies on the `f` callback raising `Not_found`. But that can probably be solved via documentation.